core/src/main/scala/cats/Foldable.scala

```diff
@@ -578,6 +578,15 @@ import Foldable.sentinel
 
   override def unorderedFoldMap[A, B: CommutativeMonoid](fa: F[A])(f: (A) => B): B =
     foldMap(fa)(f)
+
+  def contains[A](fa: F[A])(v: A)(implicit ev: Eq[A]): Boolean =
+    exists(fa)(ev.eqv(_, v))
+
+  def foldSmash[A](fa: F[A])(prefix: A, delim: A, suffix: A)(implicit M: Monoid[A]): A =
+    M.combine(prefix, M.combine(intercalate(fa, delim), suffix))
+
+  def mkString[A: Monoid](fa: F[A])(prefix: String, delim: String, suffix: String)(implicit F: Functor[F], S: Show[A], ev1: Monoid[String]): String =
```

Comment:
Hey @rsoeldner,

Actually, for `mkString` I believe we can also generalize over `String`, so `foldSmash` would work with `String`. Something like:

```scala
def mkString[A: Monoid, B: Show](fa: F[A])(prefix: B, delim: B, suffix: B)(implicit F: Functor[F], S: Show[A]): String =
  foldSmash(F.map(fa)(S.show))(prefix.show, delim.show, suffix.show)
```

Maybe this is bit a bike-shedding, but probably can help if you want to have unusual `Show[B]` instance in scope.

Reply:
Right, I thought about this, but wasn't able to find a use-case. Do you have an Idea ?

Reply:
I guess the main question here is "why I might want to pass different `B`s into `mkString`". I think it has use-case for [pretty printers](https://github.com/typelevel/paiges) to output some structure differently based on environment:

```
structure.mkString(LeftPad(0), LeftPad(5), LeftPad(0))
```

Maybe this is a completely made-up example, but why to constrain it?

Reply:
I see your point, should we support different types for _prefix_, _suffix_ and _delim_ ?

Reply:
Don't know, for me it looks like a skew into "over-generalized" approach, but let's see what community says - maybe even suggestion with `Show[B]` is unnecessary.

Reply:
:+1:

Reply:
we shouldn't need `Functor[F]`. You can do this 
```scala
def mkString[A:Show, B: Show](fa: F[A])(prefix: B, delim: B, suffix: B): String =
  foldLeft(fa, prefix.show) { (left, a) =>
    left + (delim.show) + a.show
  } + (suffix.show)
```
of course, that is a O(N^2) implementation, but it could be fixed using a `StringBuilder`:
```scala
def mkString[A:Show, B: Show](fa: F[A])(prefix: B, delim: B, suffix: B): String = {
  val bldr = new java.lang.StringBuilder
  bldr.append(prefix.show)
  val b2 = foldLeft(fa, bldr) { (b, a) =>
    b.append((delim.show) + a.show)
  }
  b2.append(suffix.show)
  b2.toString
```

Reply:
that said, I think `B` is overgeneralized. Users should just do `prefix.show` etc... and keep it to be `String`.

Reply:
After some time, I think you are right. :+1: